lib/portlayer/exec/exec.go

```diff
@@ -161,69 +155,20 @@ func eventCallback(ie events.Event) {
 				}()
 			case StateRemoved:
 				log.Debugf("Container(%s) %s via event activity", container.ExecConfig.ID, newState.String())
+				if container.vm != nil && container.vm.Fixing {
+					// is fixing vm, do not remove from containers
+					log.Debugf("Container(%s) %s is being fixed", container.ExecConfig.ID)
+					container.vm.Fixing = false
+					break
+				}
 				Containers.Remove(container.ExecConfig.ID)
 				publishContainerEvent(container.ExecConfig.ID, ie.Created(), ie.String())
-
 			}
 		}
 	}
 	return
 }
 
-// registeredVMCallback will process registeredVMEvent
-func registeredVMCallback(sess *session.Session, ie events.Event) {
```

Comment:
We don't remove vm for while it's unregistered, so no need to listen for registered vm event